src/Microsoft.VisualStudio.Threading.Tests.Shared/JoinableTaskTests.cs

```diff
@@ -3249,6 +3249,34 @@ public void PostToUnderlyingSynchronizationContextShouldBeAfterSignalJoinableTas
             });
         }
 
+        [StaFact]
+        public void JoinAsyncShouldCompleteWithoutUIThreadAfterCancellation()
+        {
+            var jt = this.asyncPump.RunAsync(async delegate {
+                await Task.Yield();
+            });
+            var cts = new CancellationTokenSource();
+            var joinTask = jt.JoinAsync(cts.Token);
+            cts.Cancel();
```

Comment:
What if the joinTask has already completed (because jt's completed) by the time cts is cancelled? Then cancellation won't throw an exception. Should there be an delay in jt, or something that blocks it from completion?